libensemble/gen_classes/aposmm.py

```diff
@@ -34,88 +34,79 @@ def __init__(
             persis_info = add_unique_random_streams({}, 2, seed=4321)[1]
         super().__init__(History, persis_info, gen_specs, libE_info, **kwargs)
         if not self.persis_info.get("nworkers"):
-            self.persis_info["nworkers"] = gen_specs["user"]["max_active_runs"]  # ??????????
+            self.persis_info["nworkers"] = kwargs.get("nworkers", gen_specs["user"]["max_active_runs"])
         self.all_local_minima = []
         self._ask_idx = 0
         self._last_ask = None
         self._tell_buf = None
         self._n_buffd_results = 0
-        self._n_total_results = 0
         self._told_initial_sample = False
 
     def _slot_in_data(self, results):
         """Slot in libE_calc_in and trial data into corresponding array fields. *Initial sample only!!*"""
-        self._tell_buf["f"][self._n_buffd_results] = results["f"]
-        self._tell_buf["x"][self._n_buffd_results] = results["x"]
-        self._tell_buf["sim_id"][self._n_buffd_results] = results["sim_id"]
-        self._tell_buf["x_on_cube"][self._n_buffd_results] = results["x_on_cube"]
-        self._tell_buf["local_pt"][self._n_buffd_results] = results["local_pt"]
-
-    @property
-    def _array_size(self):
-        """Output array size must match either initial sample or N points to evaluate in parallel."""
-        user = self.gen_specs["user"]
-        return user["initial_sample_size"] if not self._told_initial_sample else user["max_active_runs"]
-
-    @property
+        self._tell_buf[self._n_buffd_results : self._n_buffd_results + len(results)] = results
+
     def _enough_initial_sample(self):
-        """We're typically happy with at least 90% of the initial sample, or we've already told the initial sample"""
         return (
-            self._n_buffd_results >= self.gen_specs["user"]["initial_sample_size"] - 10
+            self._n_buffd_results >= int(self.gen_specs["user"]["initial_sample_size"])
         ) or self._told_initial_sample
 
+    def _ready_to_ask_genf(self):
+        """
+        We're presumably ready to be asked IF:
+        - When we're working on the initial sample:
+            - We have no _last_ask cached
+            - all points given out have returned AND we've been asked *at least* as many points as we cached
```

Comment:
I'm not sure this is necessary with my change to Optimas so APOSMM is asked all points at once. Though not sure without testing it.